includes/utils/class-users.php

```diff
@@ -53,4 +55,58 @@ public static function get_or_create_user_by_email( $email, $remote_site_url, $r
 
 	}
 
+	/**
+	 * Looks for avatar information and tries to sideload it to the user
+	 *
+	 * @param int   $user_id The user ID to add the avatar to.
+	 * @param array $user_data The user data where we are going to look for the avatar. We are looking for the 'simple_local_avatar' meta key.
+	 * @param bool  $overwrite Whether to overwrite the existing avatar or not.
+	 * @return bool True if the avatar was sideloaded, false otherwise.
+	 */
+	public static function maybe_sideload_avatar( $user_id, $user_data, $overwrite ) {
+
+		Debugger::log( 'Attempting to sideload user avatar' );
+
+		global $simple_local_avatars;
+		if ( ! $simple_local_avatars || ! is_a( $simple_local_avatars, 'Simple_Local_Avatars' ) ) {
+			Debugger::log( 'Simple Local Avatars plugin not active, skipping' );
+			return false;
+		}
+
+		$avatar_meta_key = 'simple_local_avatar';
+
+		$existing_avatar = get_user_meta( $user_id, $avatar_meta_key, true );
+
+		if ( ! empty( $existing_avatar ) && ! $overwrite ) {
+			Debugger::log( 'User already has an avatar and overwrite is false, skipping' );
+			return false;
+		}
+
+		if ( ! is_array( $user_data ) ) {
+			$user_data = (array) $user_data;
+		}
+
+		if ( array_key_exists( $avatar_meta_key, $user_data ) && ! empty( $user_data[ $avatar_meta_key ]['full'] ) ) {
+			Debugger::log( 'Updating user avatar' );
+			$avatar_url = $user_data[ $avatar_meta_key ]['full'];
+
+			if ( ! function_exists( 'media_sideload_image' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/media.php';
+				require_once ABSPATH . 'wp-admin/includes/file.php';
+				require_once ABSPATH . 'wp-admin/includes/media.php';
+				require_once ABSPATH . 'wp-admin/includes/image.php';
```

Comment:
`wp-admin/includes/media.php` is added twice

Reply:
https://github.com/Automattic/newspack-network/pull/21/commits/00c4da889eae623339396e380fbcc7ed134ca666